Core/ComponentBaseClasses/elxTransformBase.hxx

```diff
@@ -1709,7 +1712,10 @@ TransformBase<TElastix>::SetReadWriteTransformParameters(const bool _arg)
   {
     this->m_ReadWriteTransformParameters = _arg;
     ObjectType * thisAsObject = dynamic_cast<ObjectType *>(this);
-    thisAsObject->Modified();
+    if (thisAsObject != nullptr)
```

Comment:
@thewtex Thanks you, Matt, but have you actually seen `dynamic_cast<ObjectType *>(this)` returning a `nullptr`? Because I believe the cast should _always_ succeed.

Reply:
let's first make a pr without this change. If Matt actually saw a problem here it can be a separate commit, as it addresses a separate problem also.

Reply:
Yes, I was observing `nullptr` / `0x0` with lldb as described in #411. Feel free to split up this branch, rebase, etc.

Reply:
@thewtex  Ah, I see now! You got a nullptr! 🙀 It is really strange because unchecked use of this `dynamic_cast<ObjectType *>(this)` has been in there for a very long time! 14 October 2004:

https://github.com/SuperElastix/elastix/blob/396792f05becfe1dbe809bbab37d7758f1b5a835/src/Core/ComponentBaseClasses/elxTransformBase.hxx#L809-L810

I suspect that it has to do with the whole clang visibility/RTTI issue. So I hope that the `if (thisAsObject != nullptr)` is not necessary anymore once the visibility is fixed.

Reply:
> So I hope that the if (thisAsObject != nullptr) is not necessary anymore once the visibility is fixed.

Yes. There are many other classes in the elastix codebase that need the `ITK_TEMPLATE_EXPORT` macro. Until the visibility source is unearthed, we may want to patch the version used in ITKElastix to avoid crashing on macOS.